x/twap/api_test.go

```diff
@@ -76,26 +81,43 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 		return getTwapInput{1, quoteAssetDenom, baseAssetDenom, startTime, endTime}
 	}
 
+	makeSimpleTwapToNowInput := func(startTime time.Time, isQuoteTokenA bool) getTwapInput {
+		quoteAssetDenom, baseAssetDenom := denom0, denom1
+		if isQuoteTokenA {
+			baseAssetDenom, quoteAssetDenom = quoteAssetDenom, baseAssetDenom
+		}
+		return getTwapInput{1, quoteAssetDenom, baseAssetDenom, startTime, startTime}
+	}
+
 	quoteAssetA := true
 	quoteAssetB := false
-	// base record is a record with t=baseTime, sp0=10, sp1=.1, accumulators set to 0
+	// base record is a record with t=baseTime, sp0=10(sp1=0.1) accumulators set to 0
 	baseRecord := newTwapRecordWithDefaults(baseTime, sdk.NewDec(10), sdk.ZeroDec(), sdk.ZeroDec())
-	// record with t=baseTime+10, sp0=5, sp1=.2, accumulators updated from baseRecord
-	// accum0 = 10 seconds * (spot price = 10), accum1 = 10 seconds * (spot price = .1)
+
+	// accum0 = 10 seconds * (spot price = 10) = OneSec * 10 * 10
+	// accum1 = 10 seconds * (spot price = 0.1) = OneSec
 	accum0, accum1 := OneSec.MulInt64(10*10), OneSec
+
+	// accumulators updated from baseRecord with
+	// t = baseTime + 10
+	// sp0 = 5, sp1 = 0.2
 	tPlus10sp5Record := newTwapRecordWithDefaults(
 		baseTime.Add(10*time.Second), sdk.NewDec(5), accum0, accum1)
-	// TODO: Make use of the below for test cases:
-	// record with t=baseTime+20, sp0=2, sp1=.5, accumulators updated from tPlus10sp5Record
-	// tPlus20sp2Record := newTwapRecordWithDefaults(
-	// 	baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
+
+	// accumulators updated from tPlus10sp5Record with
+	// t = baseTime + 20
+	// sp0 = 2, sp1 = 0.5
+	tPlus20sp2Record := newTwapRecordWithDefaults(
+		baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
 
 	tests := map[string]struct {
 		recordsToSet []types.TwapRecord
 		ctxTime      time.Time
 		input        getTwapInput
 		expTwap      sdk.Dec
 		expErrorStr  string
+		// we test `GetArithmeticTwapToNow` if this field is set true
+		testGetArithmeticTwapToNow bool
```

Comment:
Can we have a separate test fixture for this please so that each one is focused on testing one thing?

I know that the methods are similar but I think that separating them in tests would still make both more focused and readable

Reply:
yeah I was also thinking if I should be separating them out or not. 
I could be doing that, but 200+ lines of the test codes would be basically be duplicated. Regarding the fact that `GetArithmeticTwapToNow` does not have any additional functionality from `GetArithmeticTwap` besides the fact that we use the current time as its parameter, I personally don't think its worth the tradeoff for all unnecessary code duplication in the test code. Lmk what you think!